cartridge/roles/coordinator.lua

```diff
@@ -247,7 +252,22 @@ local function apply_config(conf, _)
         return true
     end
 
-    if failover_cfg.state_provider ~= 'tarantool' then
+    if failover_cfg.state_provider == 'tarantool' then
+        local params = assert(failover_cfg.tarantool_params)
+
+        if vars.client == nil
```

Comment:
It would be great if you find a way to generalise such logic. You change it here if ~something~ is changed. I suppose you could organise configuration in a way then `cmpdeeply(new, old)` makes you call client:stop/drop and reconnect then.